app/scripts/lib/typed-message-manager.js

```diff
@@ -313,6 +314,17 @@ export default class TypedMessageManager extends EventEmitter {
     this._setMsgStatus(msgId, 'errored');
   }
 
+  /**
+   * Clears all unapproved messages from memory.
+   */
+  clearUnapproved() {
+    this.memStore.updateState({
```

Comment:
For some ungodly reason we store signature requests in _two_ different places in these message manager classes. They are communicated to the UI via this `memStore` but the primary list is in `this.messages`, which remains unchanged here. Take a look at `getUnapprovedMsgs` for example - that would still return messages "cleared" here.

I believe the same applies to the other message manager classes as well.

Reply:
Huh, I don't disagree, but I'm very curious how this was working in practice then. I'll dig into that to understand more and submit this fix in the meantime